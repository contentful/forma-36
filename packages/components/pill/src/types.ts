export type PillVariants = 'idle' | 'active' | 'deleted';
