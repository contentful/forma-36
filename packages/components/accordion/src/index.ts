export { Accordion } from './Accordion';
export type { AccordionProps } from './Accordion';
export { AccordionHeader } from './AccordionHeader/AccordionHeader';
export type { AccordionHeaderProps } from './AccordionHeader/AccordionHeader';
export { AccordionItem } from './AccordionItem/AccordionItem';
export type { AccordionItemProps } from './AccordionItem/AccordionItem';
export { AccordionPanel } from './AccordionPanel/AccordionPanel';
export type { AccordionPanelProps } from './AccordionPanel/AccordionPanel';
