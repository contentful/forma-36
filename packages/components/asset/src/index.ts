export { Asset } from './Asset';
export type { AssetProps } from './Asset';
