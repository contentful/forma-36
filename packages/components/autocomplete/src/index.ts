export { Autocomplete } from './Autocomplete';
export type { AutocompleteProps } from './Autocomplete';
