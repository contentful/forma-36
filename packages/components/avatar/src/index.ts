export { Avatar } from './Avatar';
export type { AvatarProps } from './Avatar';
export { AvatarGroup } from './AvatarGroup';
export type { AvatarGroupProps } from './AvatarGroup';
