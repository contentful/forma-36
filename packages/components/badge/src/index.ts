export { Badge } from './Badge';
export type { BadgeProps } from './Badge';
export { EntityStatusBadge } from './EntityStatusBadge';
export type { EntityStatusBadgeProps } from './EntityStatusBadge';
export type { BadgeVariant } from './types';
