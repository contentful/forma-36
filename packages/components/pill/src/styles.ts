import { css } from 'emotion';
import tokens from '@contentful/f36-tokens';
import { PillVariants } from './types';
import type { CSSObject } from '@emotion/serialize';

const getCloseButtonStyle = (variant: PillVariants) => {
  const variantStyle = (variant: PillVariants): CSSObject => {
    switch (variant) {
      case 'active':
        return {
          borderLeft: `1px solid ${tokens.gray400}`,
          '&:hover, &:focus': {
            background: tokens.gray300,
            cursor: 'pointer',
          },
        };
      default:
        return {
          borderLeft: `1px solid ${tokens.gray400}`,
          '&:hover, &:focus': {
            background: tokens.gray300,
            cursor: 'pointer',
          },
        };
    }
  };

  return css({
    border: 'none',
    borderTopRightRadius: tokens.borderRadiusSmall,
    borderBottomRightRadius: tokens.borderRadiusSmall,
    padding: tokens.spacingXs,
    background: 'transparent',
    transition: `background ${tokens.transitionDurationShort} ${tokens.transitionEasingDefault}`,
    ...variantStyle(variant),
  });
};

const getLabelStyle = () => {
  return css({
    color: tokens.gray700,
    lineHeight: tokens.lineHeightM,
    padding: tokens.spacingXs,
    fontSize: tokens.fontSizeM,
    flexGrow: 2,
    whiteSpace: 'nowrap',
    textOverflow: 'ellipsis',
    overflow: 'hidden',
  });
};

const icon = {
  fill: tokens.gray600,
  verticalAlign: 'middle',
};

const dragIcon = {
  padding: tokens.spacingXs,
  paddingRight: 0,
};

const getPillStyle = (variant: PillVariants) => {
  const variantStyle = (variant: PillVariants) => {
    switch (variant) {
      case 'active':
        return {
          background: tokens.gray300,
        };
      case 'deleted':
        return {
          background: tokens.gray200,
          textDecoration: 'line-through',
          opacity: '0.5',
        };
      default:
        return {
          background: tokens.gray200,
        };
    }
  };

  return css({
    display: 'inline-flex',
    fontFamily: tokens.fontStackPrimary,
    alignItems: 'center',
    borderRadius: tokens.borderRadiusSmall,
    border: 'none',
    padding: 0,
    maxWidth: '100%',
    ...variantStyle(variant),
  });
};

export default (variant: PillVariants) => {
  return {
    closeButton: getCloseButtonStyle(variant),
    label: getLabelStyle(),
    icon: css(icon),
    dragIcon: css(dragIcon),
    pill: getPillStyle(variant),
  };
};
