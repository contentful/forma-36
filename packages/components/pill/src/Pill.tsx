import React from 'react';
import { cx } from 'emotion';
import type {
  CommonProps,
  PropsWithHTMLElement,
  ExpandProps,
} from '@contentful/f36-core';
import { Tooltip } from '@contentful/f36-tooltip';
import { DragIcon, CloseIcon } from '@contentful/f36-icons';
import { Button } from '@contentful/f36-button';
import { PillVariants } from './types';
import { getPillStyles } from './Pill.styles';

export type PillInternalProps = CommonProps & {
  /**
   * Text that will be shown on the pill
   */
  label: string;
  /**
   * Function that handles when the close icon is clicked. Close icon visibility depends on if this property is set.
   */
  onClose?: () => void;
  /**
   * Function that handles when the pill is dragged. Drag icon visibility depends on if this property is set.
   */
  onDrag?: () => void;
  /**
   * Custom component to be used as handler for the drag functionality.
   */
  dragHandleComponent?: React.ReactNode;
  /**
   * Determines style variation of Pill component
   * @default idle
   */
  variant?: PillVariants;
};

export type PillProps = PropsWithHTMLElement<PillInternalProps, 'div'>;

export const Pill = React.forwardRef<HTMLDivElement, ExpandProps<PillProps>>(
  (props, ref) => {
    const {
      label,
      onClose,
      testId = 'cf-ui-pill',
      onDrag,
      className,
      dragHandleComponent,
      variant = 'idle',
      ...otherProps
    } = props;

    const styles = getPillStyles(variant);
    const [textIsTruncated, setTextIsTruncated] = React.useState(false);

    const trackRefChange = React.useCallback(
      (ref: HTMLDivElement | null) => {
        if (!ref) {
          return;
        }
        const { scrollWidth, offsetWidth } = ref.parentElement;
        setTextIsTruncated(scrollWidth > offsetWidth);
      },
      [setTextIsTruncated],
    );

    return (
      <div
        className={cx(styles.pill, className)}
        data-test-id={testId}
        onDrag={onDrag}
        ref={ref}
        {...otherProps}
      >
        {onDrag &&
          (dragHandleComponent ? (
            dragHandleComponent
          ) : (
            <span aria-label="Drag handler" className={styles.dragIcon}>
              <DragIcon className={styles.icon} variant="muted" />
            </span>
          ))}
        <Tooltip
          content={label}
          maxWidth="none"
          targetWrapperClassName={styles.label}
          isDisabled={!textIsTruncated}
        >
          <span ref={trackRefChange}>{label}</span>
        </Tooltip>
        {onClose && (
          <Button
            type="button"
            variant="transparent"
            startIcon={<CloseIcon aria-label="Close" />}
            aria-label="Close"
            onClick={onClose}
            className={styles.closeButton}
          />
        )}
      </div>
    );
  },
);

Pill.displayName = 'Pill';
