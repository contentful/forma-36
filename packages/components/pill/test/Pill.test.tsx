import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { axe } from '@/scripts/test/axeHelper';
import tokens from '@contentful/f36-tokens';

import { Pill } from '../src/Pill';

describe('Pill', () => {
  it('renders the component', () => {
    const { getByText } = render(<Pill label="test" />);

    expect(getByText('test')).toBeTruthy();
  });

  it('renders the component with an additional class name', () => {
    const { container } = render(
      <Pill className="my-extra-class" label="test" />,
    );

    expect(container.firstChild).toHaveClass('my-extra-class');
  });

  it('renders the component with a dragging handle', () => {
    const mockOnDrag = jest.fn();
    const { getByLabelText, container } = render(
      <Pill label="test" onDrag={mockOnDrag} />,
    );

    const dragHandler = getByLabelText('Drag handler');
    expect(dragHandler).toBeTruthy();
    fireEvent.drag(container.firstChild);
    expect(mockOnDrag).toHaveBeenCalled();
  });

  it('renders the component with a close button', () => {
    const mockOnClose = jest.fn();
    const { container, getByRole } = render(
      <Pill label="test" onClose={mockOnClose} />,
    );

    const button = getByRole('button');
    expect(container.firstChild).toContainElement(button);
    fireEvent.click(button);
    expect(mockOnClose).toHaveBeenCalled();
  });

  it('renders the component in "idle" variant by default', () => {
    const { container } = render(<Pill label="test" />);

    expect(container.firstChild).toHaveStyle({
      background: tokens.gray200,
    });
  });

  it('renders the component in "active" variant', () => {
    const { container } = render(<Pill label="test" variant="active" />);

    expect(container.firstChild).toHaveStyle({
      background: tokens.gray300,
    });
  });

  it('renders the component in "deleted" variant', () => {
    const { container } = render(<Pill label="test" variant="deleted" />);

    expect(container.firstChild).toHaveStyle({
      background: tokens.gray200,
      textDecoration: 'line-through',
      opacity: '0.5',
    });
  });

  it('renders the component with a test id', () => {
    const { getByTestId } = render(<Pill label="test" testId="test-id" />);

    expect(getByTestId('test-id')).toBeTruthy();
  });

  it('has no a11y issues', async () => {
    const { container } = render(<Pill label="test" />);
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});
