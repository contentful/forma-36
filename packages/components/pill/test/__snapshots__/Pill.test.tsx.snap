// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders no close button in "deleted" variant 1`] = `
.emotion-3 {
  border: none;
  border-top-right-radius: 4px;
  border-bottom-right-radius: 4px;
  padding: 0.5rem;
  background: transparent;
  -webkit-transition: background 0.1s ease-in-out;
  transition: background 0.1s ease-in-out;
  border-left: 1px solid #AEC1CC;
}

.emotion-3:hover,
.emotion-3:focus {
  background: #CFD9E0;
  cursor: pointer;
}

.emotion-0 {
  color: #414D63;
  line-height: 1.25rem;
  padding: 0.5rem;
  font-size: 0.875rem;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.emotion-1 {
  display: inline-block;
  fill: #5A657C;
  height: 18px;
  width: 18px;
  fill: #5A657C;
  vertical-align: middle;
}

.emotion-2 {
  display: inherit;
  outline: 0;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  border: none;
  padding: 0;
  max-width: 100%;
  background: #E7EBEE;
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
  opacity: 0.5;
}

<div
  class="my-extra-class emotion-4"
  data-test-id="cf-ui-pill"
>
  <span
    class="emotion-0"
    title="test"
  >
    test
  </span>
  <button
    aria-label="close"
    class="emotion-3"
    type="button"
  >
    <span
      class="emotion-2"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        class="emotion-1"
        data-test-id="cf-ui-icon"
        viewBox="0 0 24 24"
      >
        <path
          d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`renders the component 1`] = `
.emotion-0 {
  color: #414D63;
  line-height: 1.25rem;
  padding: 0.5rem;
  font-size: 0.875rem;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  border: none;
  padding: 0;
  max-width: 100%;
  background: #E7EBEE;
}

<div
  class="emotion-1"
  data-test-id="cf-ui-pill"
>
  <span
    class="emotion-0"
    title="test"
  >
    test
  </span>
</div>
`;

exports[`renders the component in "active" variant 1`] = `
.emotion-0 {
  color: #414D63;
  line-height: 1.25rem;
  padding: 0.5rem;
  font-size: 0.875rem;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  border: none;
  padding: 0;
  max-width: 100%;
  background: #CFD9E0;
}

<div
  class="my-extra-class emotion-1"
  data-test-id="cf-ui-pill"
>
  <span
    class="emotion-0"
    title="test"
  >
    test
  </span>
</div>
`;

exports[`renders the component in "deleted" variant 1`] = `
.emotion-0 {
  color: #414D63;
  line-height: 1.25rem;
  padding: 0.5rem;
  font-size: 0.875rem;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  border: none;
  padding: 0;
  max-width: 100%;
  background: #E7EBEE;
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
  opacity: 0.5;
}

<div
  class="my-extra-class emotion-1"
  data-test-id="cf-ui-pill"
>
  <span
    class="emotion-0"
    title="test"
  >
    test
  </span>
</div>
`;

exports[`renders the component in "idle" variant by default 1`] = `
.emotion-0 {
  color: #414D63;
  line-height: 1.25rem;
  padding: 0.5rem;
  font-size: 0.875rem;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  border: none;
  padding: 0;
  max-width: 100%;
  background: #E7EBEE;
}

<div
  class="my-extra-class emotion-1"
  data-test-id="cf-ui-pill"
>
  <span
    class="emotion-0"
    title="test"
  >
    test
  </span>
</div>
`;

exports[`renders the component with a close button 1`] = `
.emotion-3 {
  border: none;
  border-top-right-radius: 4px;
  border-bottom-right-radius: 4px;
  padding: 0.5rem;
  background: transparent;
  -webkit-transition: background 0.1s ease-in-out;
  transition: background 0.1s ease-in-out;
  border-left: 1px solid #AEC1CC;
}

.emotion-3:hover,
.emotion-3:focus {
  background: #CFD9E0;
  cursor: pointer;
}

.emotion-0 {
  color: #414D63;
  line-height: 1.25rem;
  padding: 0.5rem;
  font-size: 0.875rem;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  border: none;
  padding: 0;
  max-width: 100%;
  background: #E7EBEE;
}

.emotion-1 {
  display: inline-block;
  fill: #5A657C;
  height: 18px;
  width: 18px;
  fill: #5A657C;
  vertical-align: middle;
}

.emotion-2 {
  display: inherit;
  outline: 0;
}

<div
  class="my-extra-class emotion-4"
  data-test-id="cf-ui-pill"
>
  <span
    class="emotion-0"
    title="test"
  >
    test
  </span>
  <button
    aria-label="close"
    class="emotion-3"
    type="button"
  >
    <span
      class="emotion-2"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        class="emotion-1"
        data-test-id="cf-ui-icon"
        viewBox="0 0 24 24"
      >
        <path
          d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`renders the component with a dragging handle 1`] = `
.emotion-2 {
  color: #414D63;
  line-height: 1.25rem;
  padding: 0.5rem;
  font-size: 0.875rem;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.emotion-1 {
  padding: 0.5rem;
  padding-right: 0;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  border: none;
  padding: 0;
  max-width: 100%;
  background: #E7EBEE;
}

.emotion-0 {
  display: inline-block;
  fill: #5A657C;
  height: 18px;
  width: 18px;
  fill: #5A657C;
  vertical-align: middle;
}

<div
  class="my-extra-class emotion-3"
  data-test-id="cf-ui-pill"
>
  <span
    aria-label="Drag handler"
    class="emotion-1"
  >
    <svg
      aria-hidden="true"
      class="emotion-0"
      data-test-id="cf-ui-icon"
      viewBox="0 0 24 24"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
      />
    </svg>
  </span>
  <span
    class="emotion-2"
    title="test"
  >
    test
  </span>
</div>
`;

exports[`renders the component with a test id 1`] = `
.emotion-0 {
  color: #414D63;
  line-height: 1.25rem;
  padding: 0.5rem;
  font-size: 0.875rem;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  border: none;
  padding: 0;
  max-width: 100%;
  background: #E7EBEE;
}

<div
  class="my-extra-class emotion-1"
  data-test-id="test-id"
>
  <span
    class="emotion-0"
    title="test"
  >
    test
  </span>
</div>
`;

exports[`renders the component with an additional class name 1`] = `
.emotion-0 {
  color: #414D63;
  line-height: 1.25rem;
  padding: 0.5rem;
  font-size: 0.875rem;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  border: none;
  padding: 0;
  max-width: 100%;
  background: #E7EBEE;
}

<div
  class="my-extra-class emotion-1"
  data-test-id="cf-ui-pill"
>
  <span
    class="emotion-0"
    title="test"
  >
    test
  </span>
</div>
`;
